rest_framework/serializers.py

```diff
@@ -441,6 +441,30 @@ def run_validation(self, data=empty):
 
         return value
 
+    def _read_only_defaults(self):
+        fields = [
+            field for field in self.fields.values()
+            if (field.read_only) and (field.default != empty) and (field.source != '*') and ('.' not in field.source)
+        ]
+
+        defaults = OrderedDict()
+        for field in fields:
+            try:
+                default = field.get_default()
+            except SkipField:
+                continue
+            defaults[field.field_name] = default
```

Comment:
if we are using a field with field_name different from the field in model that is used as source then this is failing and it should be 

`defaults[field.source] = default`

Reply:
I somehow missed your comment on this, but you're correct. The code above specifically checks for a compatible field source, but uses the serializer `field_name` instead of the `source` model field.

Reply:
i had reported this and tried to work on it, but couldnt continue
and along with this issue other issue is there defined in the Issue below
https://github.com/encode/django-rest-framework/pull/7005
https://github.com/encode/django-rest-framework/issues/7003

Reply:
Hi @anveshagarwal. I'm currently following up on that PR. That's what prompted my comment. 😄

Reply:
For posterity's sake, my reasoning above was slightly wrong. The `source` checks are an indicator that we should be operating on `field.source` instead of `field.field_name`, however the way to determine this is by looking at `run_validation`. Before it calls `run_validators` (then `_read_only_defaults`), it calls `to_internal_value`. At this point, the data has been converted to its source structure. So, it makes sense that the read-only defaults would also map to its source structure, instead of the raw field names.